ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -214,8 +214,14 @@ def start_subprocess(full_cmd):
     for filename in files:
         report[filename] = []
 
-    error_re = re.compile('(/.*?\\.(?:%s)):(\\d+):(\\d+): (?:warning:|error:)' %
-                          '|'.join(extensions))
+    # Regex explanation:
+    #   \s*\^?                   : Capture many spaces, then an optional `^`. clang_tidy output isn't great.
+    #   (\/home\/^[^:]*)         : Group capture. Everything from `/home/` up until a `:`.
+    #   (\d+)                    : Group capture. Grabs line number.
+    #   (\d+)                    : Group capture. Grabs column number.
+    #   (?:warning:|error:note:) : Non-capturing group. Matches warning, error, note.
```

Comment:
```suggestion
    #   (?:warning:|error:|note:) : Non-capturing group. Matches warning, error, note.
```

Typo in the comment.